tests/test_grading.py

```diff
@@ -180,3 +183,75 @@ def test_scores_with_generate(self, generate=False):
+                                   grader=self.staff1)
+                    db.session.add(scores)
+        db.session.commit()
+
+        # Check that by default scores are hidden
+        def check_visible_scores(user, assignment, hidden=(), visible=()):
+            self.login(user.email)
+            endpoint = '/{}/'.format(assignment.name)
+            r = self.client.get(endpoint)
+            self.assert_200(r)
+            s = r.get_data().decode("utf-8")
+            for score in hidden:
+                self.assertFalse("{}:".format(score) in s)
+            for score in visible:
+                self.assertTrue("{}:".format(score) in s)
+
+        for user, assign in zip(users, self.active_assignments):
+            check_visible_scores(user, assign, hidden=['Total', 'Composition'])
+
+        # Lab assistants and students cannot make changes
+        endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment.id)
+        for email in [self.lab_assistant1.email, self.user1.email]:
+            self.login(email)
+            response = self.client.post(endpoint, data={})
+            self.assertStatus(response, 302)
+
+
+        self.login(self.staff1.email)        
+        response = self.client.post(endpoint, data={})
+        self.assert_200(response)
+        for user in users:
+            check_visible_scores(user, self.assignment, hidden=['Composition'], visible=['Total'])
+            check_visible_scores(user, self.assignment2, hidden=['Total', 'Composition'])
+
+        self.login('okadmin@okpy.org')
+        endpoint ='/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment2.id)
+        response = self.client.post(endpoint, data={'grades':'Composition'})
+        self.assert_200(response)
+        for user in users:
+            check_visible_scores(user, self.assignment, hidden=['Composition'], visible=['Total'])
+            check_visible_scores(user, self.assignment2, hidden=['Total'], visible=['Composition'])
+
+        self.login(self.staff1.email)
+        endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment.id)
+        response = self.client.post(endpoint, data={'hide':True})
+        self.assert_200(response)
+        for user in users:
+            check_visible_scores(user, self.assignment, hidden=['Total', 'Composition'])
+            check_visible_scores(user, self.assignment2, hidden=['Total'], visible=['Composition'])
+
+        self.login(self.staff1.email)
+        endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment2.id)
+        response = self.client.post(endpoint, data={})
+        self.assert_200(response)
+        for user in users:
+            check_visible_scores(user, self.assignment, hidden=['Total', 'Composition'])
+            check_visible_scores(user, self.assignment2, visible=['Composition', 'Total'])
+
```

Comment:
It's not on the student side - It would just for the Admin side grading view (which doesn't use visible_grade right now) 

That test isn't actually specific to this PR - so we'll just add that test later.